packages/spacecat-shared-data-access/src/models/configuration.js

```diff
@@ -17,14 +17,145 @@ const Configuration = (data = {}) => {
+    if (self.handlers[type]?.enabled) {
+      return self.handlers[type].enabled.orgs.includes(org.getId());
+    } else if (self.handlers[type]?.disabled) {
+      return !self.handlers[type].disabled.orgs.includes(org.getId());
+    }
+    return self.handlers[type].enabledByDefault;
+  };
+
+  self.enableHandlerTypeForSite = (type, site) => {
+    const isEnabled = self.isHandlerTypeEnabledForSite(type, site);
+    if (isEnabled) {
+      return;
+    }
+    if (self.handlers[type]?.enabledByDefault) {
+      if (self.handlers[type]?.disabled?.sites) {
+        self.handlers[type].disabled.sites = self.handlers[type].disabled.sites.filter(
+          (id) => id !== site.getId(),
+        );
+      } else {
+        self.handlers[type].disabled = { sites: [site.getId()] };
+      }
+    } else if (self.handlers[type]?.enabled?.sites) {
+      self.handlers[type]?.enabled.sites.push(site.getId());
+    } else {
+      self.handlers[type].enabled = { sites: [site.getId()] };
+    }
+  };
+
+  self.enableHandlerTypeForOrg = (type, org) => {
+    const isEnabled = self.isHandlerTypeEnabledForOrg(type, org);
+    if (isEnabled) {
+      return;
+    }
+    if (self.handlers[type]?.enabledByDefault) {
+      if (self.handlers[type]?.disabled?.orgs) {
+        self.handlers[type].disabled.orgs = self.handlers[type]?.disabled.orgs.filter(
+          (id) => id !== org.getId(),
+        );
+      } else {
+        self.handlers[type].disabled = { orgs: [org.getId()] };
+      }
+    } else if (self.handlers[type]?.enabled?.orgs) {
+      self.handlers[type]?.enabled.orgs.push(org.getId());
+    } else {
+      self.handlers[type].enabled = { orgs: [org.getId()] };
+    }
+  };
+
+  self.disableHandlerTypeForSite = (type, site) => {
+    const isEnabled = self.isHandlerTypeEnabledForSite(type, site);
+    if (!isEnabled) {
+      return;
+    }
+    if (self.handlers[type]?.enabledByDefault) {
+      if (self.handlers[type]?.enabled?.sites) {
+        self.handlers[type].enabled.sites = self.handlers[type]?.enabled.sites.filter(
+          (id) => id !== site.getId(),
+        );
+      }
+    } else if (self.handlers[type]?.disabled?.sites) {
+      self.handlers[type]?.disabled.sites.push(site.getId());
+    } else {
+      self.handlers[type].disabled = { sites: [site.getId()] };
+    }
+  };
+
+  self.disableHandlerTypeForOrg = (type, org) => {
+    const isEnabled = self.isHandlerTypeEnabledForOrg(type, org);
+    if (!isEnabled) {
+      return;
+    }
+    if (self.handlers[type]?.enabledByDefault) {
+      if (self.handlers[type]?.enabled?.orgs) {
+        self.handlers[type].enabled.orgs = self.handlers[type]?.enabled.orgs.filter(
+          (id) => id !== org.getId(),
+        );
+      }
+    } else if (self.handlers[type]?.disabled?.orgs) {
+      self.handlers[type]?.disabled.orgs.push(org.getId());
+    } else {
+      self.handlers[type].disabled = { orgs: [org.getId()] };
+    }
+  };
```

Comment:
create utility function and use in your methods to shorten and make more readable:
```suggestion
const updateHandlerOrgs = (type, orgId, enabled) => {
  const handler = self.handlers[type];
  if (!handler) return;

  if (enabled) {
    if (handler.enabledByDefault) {
      handler.disabled.orgs = handler.disabled.orgs?.filter(id => id !== orgId) || [];
    } else {
      handler.enabled.orgs = [...(handler.enabled.orgs || []), orgId];
    }
  } else {
    if (handler.enabledByDefault) {
      handler.enabled.orgs = handler.enabled.orgs?.filter(id => id !== orgId) || [];
    } else {
      handler.disabled.orgs = [...(handler.disabled.orgs || []), orgId];
    }
  }
};

self.enableHandlerTypeForSite = (type, site) => {
  const siteId = site.getId();
  if (self.isHandlerTypeEnabledForSite(type, site)) return;

  updateHandlerSites(type, siteId, true);
};

self.enableHandlerTypeForOrg = (type, org) => {
  const orgId = org.getId();
  if (self.isHandlerTypeEnabledForOrg(type, org)) return;

  updateHandlerOrgs(type, orgId, true);
};

self.disableHandlerTypeForSite = (type, site) => {
  const siteId = site.getId();
  if (!self.isHandlerTypeEnabledForSite(type, site)) return;

  updateHandlerSites(type, siteId, false);
};

self.disableHandlerTypeForOrg = (type, org) => {
  const orgId = org.getId();
  if (!self.isHandlerTypeEnabledForOrg(type, org)) return;

  updateHandlerOrgs(type, orgId, false);
};
```